src/plugin_host.jl

```diff
@@ -0,0 +1,174 @@
+start_host() = try wait(nvim_child(HostHandler()).reader) end
+
+immutable HostHandler
+    loaded_plugins::Set{String} # plugin file paths
+    proc_callbacks::Dict{String, Function}
+end
+
+function HostHandler()
+    HostHandler(Set{String}(), Dict{String, Function}())
+end
+
+# names/methods for invoked cmds/fns have the form:
+#   <file path>:(command|function|autocmd):<procedure name>
+function on_notify(h::HostHandler, c, name::String, args::Vector{Any})
+    proc = require_plug(h, name)
+
+    if proc == nothing
+        throw(ErrorException(name * " not defined"))
+    end
+
+    @async(try
+        proc(c, args...)
+    catch
+        throw(MethodError(name, tuple(NvimClient, args...)))
+        throw(err)
+    end)
+end
+
+function on_request(h::HostHandler, c, serial, method, args)
+    if method == "specs" # called on UpdateRemotePlugins
+        reply_result(c, serial, get_specs(args...))
+    else
+        proc = require_plug(h, method)
+
+        if proc == nothing
+            reply_error(c, serial, "Command not defined.")
+        end
+
+        @async(try
+            reply_result(c, serial, proc(c, args...))
+        catch
+            err = "No matching method for args (NvimClient, " * join(args, ", ") * ")"
```

Comment:
This could be any error, and not just  `MethodError` for `proc`. Lets use `showerror` here as well. After `reply_error`, we could also print it in `STDERR` so that all errors are in the log.